torch/quantization/fx/_equalize.py

```diff
@@ -497,21 +502,20 @@ def convert_eq_obs(
             inp_quant_obs_node = node.args[0]
             prev_node = inp_quant_obs_node.args[0]
 
-            # TODO: Possible special handling for connected linear layers
-            # We might need to replace the InputEqualizationObserver with another mul operator...
-            # if prev_node.op == 'call_module' and modules[prev_node.target].dtype != torch.float32:
-            # If this is a connecting linear layer, we want to remove the InputEqualizationObserver
-            #     # Replace the current node with the previous node
-            #     orig_users = list(node.users.keys())
-            #     for user_node in orig_users:
-            #         user_node.replace_input_with(node, prev_node)
-            #     # Erase the node
-            #     model.graph.erase_node(node)
-            #     continue
-
             # Update the following input quantization observer's min/max values
             scale_input_observer(node, modules)
 
+            if prev_node.op == 'call_module' and isinstance(prev_node.target, str) and \
+               isinstance(modules[prev_node.target], nn.Linear):
```

Comment:
Can we use current nodes' module type instead of restricting it to only nn.Linear here?

Reply:
I'm not too sure if I understand what you mean. The current node is an InputEqualizationObserver and we would like to check that 2 nodes ago it was a Linear layer. I think I could replace this line with a call to `node_supports_equalization(prev_node, modules)`, but I'm not sure how we could use the current node's module type.

Reply:
Yes, sorry for my confusing comment. I mis-understood the context this code is being called from. Yes if we can use `node_supports_equalization(prev_node, modules)` that would be better